File: BACKEND/materials_management/created_materials/controllers/MaterialController.php
<?php
require_once __DIR__ . '/../models/MaterialModel.php';

class MaterialController {
    private $model;

    public function __construct() {
        $this->model = new MaterialModel();
    }

    public function createMaterial() {
        header("Content-Type: application/json");

        $data = json_decode(file_get_contents("php://input"), true);

        if (!isset($data['name'], $data['subject'], $data['description'])) {
            echo json_encode(["error" => "Missing required fields"]);
            http_response_code(400);
            return;
        }

        $result = $this->model->createMaterial($data['name'], $data['subject'], $data['description']);
        
        echo json_encode($result);
        http_response_code(isset($result['error']) ? 500 : 201);
    }
}
?>

File: BACKEND/materials_management/created_materials/controllers/MaterialController.go
package controllers

import (
	"encoding/json"
	"net/http"

	"created_materials/db"
	"created_materials/models"
)

// CreateMaterial adds a new material to the database.
//
// @Summary Create a new material
// @Description Adds a new material entry to the database.
// @Tags materials
// @Accept json
// @Produce json
// @Param material body models.Material true "Material data"
// @Success 201 {object} map[string]string "message: Material created successfully"
// @Failure 400 {string} string "Invalid input"
// @Failure 500 {string} string "Internal server error"
// @Router /materials [post]
func CreateMaterial(w http.ResponseWriter, r *http.Request) {
	var material models.Material
	err := json.NewDecoder(r.Body).Decode(&material)
	if err != nil {
		http.Error(w, "Invalid input", http.StatusBadRequest)
		return
	}

	query := "INSERT INTO materials (name, subject, description, created_at) VALUES (?, ?, ?, NOW())"
	_, err = db.DB.Exec(query, material.Name, material.Subject, material.Description)
	if err != nil {
		http.Error(w, "Error creating material", http.StatusInternalServerError)
		return
	}

	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(map[string]string{"message": "Material created successfully"})
}
